lib/openstudio-standards/weather/stat_file.rb

```diff
@@ -342,7 +349,9 @@ def parse
         # week periods
         regex = /Typical Week Period selected:(.*?)C/
         match_data = @text.scan(regex)
-        if match_data.nil? || match_data.empty?
+        if @text.include?('No discernible dry period')
+          OpenStudio.logFree(OpenStudio::Warn, 'openstudio.Weather.stat_file', "Can't find typical weather weeks in the .stat file because it has no discernible dry period .")
+        elsif match_data.nil? || match_data.size < 2
```

Comment:
Adjusting this anyways, because you ARE doing match_data[0] and match_data[1], so it's the logical test instead of doing `match_data.empty?`